### PR TITLE
Automatically deploy documentation to docs.contao.org

### PR DESCRIPTION
**Pull request by Toflar**
This introduces automated deployment of the master branch to docs.contao.org on every push to the master branch.

### CODE REVIEW

File: .github/workflows/build_and_deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+name: Build and deploy to https://docs.contao.org
+
+on:
+    push:
+        branches:
+        - master
```
Comment (leofeyer) [inline .github/workflows/build_and_deploy.yaml]: Is this line correctly indented?

Comment (Toflar) [reply .github/workflows/build_and_deploy.yaml]: Nope :D

Comment (Toflar) [reply .github/workflows/build_and_deploy.yaml]: Fixed.

**Comment by aschempp** [comment]
Does this include multilingual manuals? @fritzmg told me the current URL is actually wrong, as it does not contain `de`.

**Comment by Toflar** [comment]
That's out of scope here. That's a build task, not a deployment task.

**Comment by fritzmg** [comment]
> Does this include multilingual manuals? @fritzmg told me the current URL is actually wrong, as it does not contain `de`.

Yes, once we re-introduce the language parameter, it will be automatically deployed as well.

**Pull request merged**